File: src/main/java/com/hazza/LeetCodeByJava/No89GrayCode.java
package com.hazza.LeetCodeByJava;

import java.util.LinkedList;
import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * Description: https://leetcode.com/problems/gray-code/description/
 * Author: HazzaCheng
 * Contact: hazzacheng@gmail.com
 * Date: 2017-09-26
 * Time: 10:25 AM
 */
public class No89GrayCode {
    public List<Integer> grayCode(int n) {
        List<Integer> res = new LinkedList<>();
        for (int i = 0; i < 1<<n; i++) res.add(i ^ i>>1);
        return res;
    }
}

File: src/main/python/bit/No89_Gray_Code.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019-07-29
# https://leetcode.com/problems/gray-code/
from typing import List


class Solution:
    def grayCode(self, n: int) -> List[int]:
        res = [0]
        for i in range(n):
            res += [x | (1 << i) for x in res[::-1]]
        return res
